forge/forge/components/git_operations/git_operations.py

```diff
@@ -1,21 +1,31 @@
 from pathlib import Path
-from typing import Iterator
+from typing import Iterator, Optional
 
 from git.repo import Repo
+from pydantic import BaseModel, SecretStr
 
+from forge.agent.components import ConfigurableComponent
 from forge.agent.protocols import CommandProvider
 from forge.command import Command, command
-from forge.config.config import Config
+from forge.models.config import FromEnv
 from forge.models.json_schema import JSONSchema
 from forge.utils.exceptions import CommandExecutionError
 from forge.utils.url_validator import validate_url
 
 
-class GitOperationsComponent(CommandProvider):
+class GitOperationsConfiguration(BaseModel):
+    github_username: Optional[str] = FromEnv("GITHUB_USERNAME")
+    github_api_key: Optional[SecretStr] = FromEnv("GITHUB_API_KEY")
+
+
+class GitOperationsComponent(
+    CommandProvider, ConfigurableComponent[GitOperationsConfiguration]
+):
     """Provides commands to perform Git operations."""
 
-    def __init__(self, config: Config):
-        self._enabled = bool(config.github_username and config.github_api_key)
+    def __init__(self, config: Optional[GitOperationsConfiguration] = None):
+        super().__init__(config or GitOperationsConfiguration())
```

Comment:
Which parent class handles this `super()` call? (I can guess but it isn't explicit)

Should either call the `__init__` of that class specifically, or use `super(GitOperationsComponent, self).__init__(config=config)` for proper multi-inheritance.

Reply:
I guess explicit is more clear? `ConfigurableComponent.__init__(self, config)`